conformance/tests/overloads_overlap.py

```diff
@@ -0,0 +1,40 @@
+"""
+Tests checks for overlapping overloads.
+"""
+
+from typing import Literal, overload
+
+# > If two overloads can accept the same set of arguments, they are said
+# > to "partially overlap". If two overloads partially overlap, the return type
+# > of the former overload should be assignable to the return type of the
+# > latter overload. If this condition doesn't hold, it is indicative of a
+# > programming error and should be reported by type checkers.
+
+@overload
+def is_one(x: Literal[1]) -> Literal[True]:  # E: overlapping overloads, inconsistent return type
+    ...
+
+@overload
+def is_one(x: int) -> Literal[False]:
+    ...
+
+def is_one(x: int) -> bool:
+    return x == 1
+
```

Comment:
I realize that conformance tests are intended to be very basic, but this seems slightly too basic. Can we add a case with at least two arguments and a case with default values? E.g. `mypy` stopped flagging the following as an error a while ago (probably due to a regression; those signature do "partially overlap" according to the definition in spec and have incompatible return types):

```python
@overload
def fn(x: str = ...) -> str: ...
@overload
def fn(x: int = ...) -> int: ...
def fn(x: int | str = 0) -> int | str:
    return x
```

Reply:
This test is no longer applicable because I've removed the section of the spec related to overlapping overloads. We weren't able to achieve consensus on this behavior.